packages/gatsby-remark-prismjs/src/highlight-line-range.js

```diff
@@ -0,0 +1,147 @@
+const rangeParser = require(`parse-numeric-range`)
+
+const COMMENT_START = /(#|\/\/|\{\/\*|\/\*+|<!--)/
+const COMMENT_END = /(-->|\*\/\}|\*\/)?/
+const DIRECTIVE = /highlight-(next-line|line|start|end|range)({([^}]+)})?/
+const END_DIRECTIVE = /highlight-end/
+const plainTextWithLFTest = /<span class="token plain-text">[^<]*\n[^<]*<\/span>/g
+
+const stripComment = line =>
```

Comment:
I think we'll have to re-visit this. Here's what happens with a JSX comment:

![screen shot 2018-11-12 at 2 19 51 pm](https://user-images.githubusercontent.com/3924690/48372997-0ba26b80-e686-11e8-9c7d-1c0c6e6d9506.png)

So it's _close_ but not quite there! Checking out the reactjs.org site now.

Reply:
Fixed in 4b80fc7 — problem was that the code that was processed in the `stripComment` function had already gone through Prism, so there was a few extra `span`s here and there.

I added this comment for clarity:

https://github.com/gatsbyjs/gatsby/blob/4b80fc7291e659e79f8f9aef209865bdd5e69656/packages/gatsby-remark-prismjs/src/highlight-line-range.js#L3-L19

Not 100% happy with this solution though, if you have an idea for a more elegant one I’d be willing to try it!